File: ComPDFKit 1.11.0 for Windows/Examples/Samples/PDFRedactTest/VB/PDFRedactTest.vb
Imports ComPDFKit.Import
Imports ComPDFKit.PDFAnnotation
Imports ComPDFKit.PDFDocument
Imports ComPDFKit.PDFPage
Imports System
Imports System.IO
Imports System.Windows

Module PDFRedactTest
    Private outputPath As String = Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(System.IO.Directory.GetCurrentDirectory()))) & "\Output\VB"

    Sub Main()
        ' Perparation work
        Console.WriteLine("Running redact test sample…" & vbCrLf)
        SDKLicenseHelper.LicenseVerify()

        Dim document As CPDFDocument = CPDFDocument.InitWithFilePath("CommonFivePage.pdf")

        Dim str As String = document.PageAtIndex(0).GetTextPage().GetSelectText(New Point(300, 240), New Point(400, 300), New Point(0, 0))
        Console.WriteLine("The text need to be redact is: {0}", str)

        If Not Directory.Exists(outputPath) Then
            Directory.CreateDirectory(outputPath)
        End If

        ' Redact
        If Redact(document) Then
            Console.WriteLine("Redact done.")
        Else
            Console.WriteLine("Redact failed.")
        End If

        Console.WriteLine("--------------------")

        Console.WriteLine("Done!")
        Console.WriteLine("--------------------")

        Console.ReadLine()
    End Sub

    ' Redact an area in PDF
    Private Function Redact(document As CPDFDocument) As Boolean
        Dim page As CPDFPage = document.PageAtIndex(0)
        Dim redactAnnot As CPDFRedactAnnotation = TryCast(page.CreateAnnot(C_ANNOTATION_TYPE.C_ANNOTATION_REDACT), CPDFRedactAnnotation)
        ' Set redact rect: cover the title
        redactAnnot.SetRect(New CRect(300, 240, 400, 300))
        ' Set overlay text: REDACTED
        redactAnnot.SetOverlayText("REDACTED")

        ' Properties of cover text
        Dim textAttribute As New CTextAttribute()
        textAttribute.FontName = "Helvetica"
        textAttribute.FontSize = 12
        Dim fontColor As Byte() = {255, 0, 0}
        textAttribute.FontColor = fontColor
        redactAnnot.SetTextDa(textAttribute)
        redactAnnot.SetTextAlignment(C_TEXT_ALIGNMENT.ALIGNMENT_LEFT)

        ' Properties of cover square
        Dim fillColor As Byte() = {255, 0, 0}
        redactAnnot.SetFillColor(fillColor)
        Dim outlineColor As Byte() = {0, 255, 0}
        redactAnnot.SetOutlineColor(outlineColor)

        redactAnnot.UpdateAp()
        document.ApplyRedaction()
        ' Save to pointed path so you can observe the effect.
        Dim path As String = outputPath & "\RedactTest.pdf"
        If Not document.WriteToFilePath(path) Then
            Return False
        End If
        Console.WriteLine("Browse the changed file in " & path)
        Dim newDocument As CPDFDocument = CPDFDocument.InitWithFilePath(path)

        ' Validation: try to get the text of the covered area
        Dim str As String = newDocument.PageAtIndex(0).GetTextPage().GetSelectText(New Point(60, 200), New Point(560, 250), New Point(0, 0))
        Console.WriteLine("Text in the redacted area is: {0}", str)
        Return True
    End Function
End Module

File: ComPDFKit for Windows/Examples/Samples/PDFRedactTest/CS/PDFRedactTest.cs
using ComPDFKit.Import;
using ComPDFKit.PDFAnnotation;
using ComPDFKit.PDFDocument;
using ComPDFKit.PDFPage;
using System; 
using System.IO; 
using System.Windows;

namespace PDFRedactTest
{
    internal class PDFRedactTest
    {
        private static string outputPath =Path.Combine(Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(System.IO.Directory.GetCurrentDirectory()))) ?? string.Empty, "Output", "CS");
        static void Main(string[] args)
        {
            #region Perparation work
            Console.WriteLine("Running redact test sample…\r\n");
            SDKLicenseHelper.LicenseVerify();

            CPDFDocument document = CPDFDocument.InitWithFilePath("CommonFivePage.pdf");
            
            string str = document.PageAtIndex(0).GetTextPage().GetSelectText(new CPoint(300, 240), new CPoint(400, 300), new CPoint(0, 0));
            Console.WriteLine("The text need to be redact is: {0}", str);

            if (!Directory.Exists(outputPath))
            {
                Directory.CreateDirectory(outputPath);
            }
            #endregion

            #region Redact
            if (Redact(document))
            {
                Console.WriteLine("Redact done.");
            }
            else
            {
                Console.WriteLine("Redact failed.");
            }

            Console.WriteLine("--------------------");
            #endregion

            Console.WriteLine("Done!");
            Console.WriteLine("--------------------");

            Console.ReadLine();
        }

        /// <summary>
        /// Redact an area in PDF
        /// </summary>
        /// <param name="document">Regular document with some text</param>
        /// <returns></returns>
        static private bool Redact(CPDFDocument document)
        {
            CPDFPage page = document.PageAtIndex(0);
            CPDFRedactAnnotation redact = page.CreateAnnot(C_ANNOTATION_TYPE.C_ANNOTATION_REDACT) as CPDFRedactAnnotation;
            //Set radact rect: cover the title
            redact.SetRect(new CRect(300, 300, 400, 240));
            //Set overlay text: REDACTED
            redact.SetOverlayText("REDACTED");

            //Properties of cover text
            CTextAttribute textAttribute = new CTextAttribute();
            textAttribute.FontName = "Helvetica";
            textAttribute.FontSize = 12;
            byte[] fontColor = { 255, 0, 0 };
            textAttribute.FontColor = fontColor;
            redact.SetTextDa(textAttribute);
            redact.SetTextAlignment(C_TEXT_ALIGNMENT.ALIGNMENT_LEFT);

            //Properties of cover square
            byte[] fillColor = { 255, 0, 0 };
            redact.SetFillColor(fillColor);
            byte[] outlineColor = { 0, 255, 0 };
            redact.SetOutlineColor(outlineColor);

            redact.UpdateAp();
            document.ApplyRedaction();
            // Save to pointed path so you can observe the effect.
            string path = Path.Combine(outputPath, "RedactTest.pdf");
            if (!document.WriteToFilePath(path))
            {
                return false;
            }
            Console.WriteLine("Browse the changed file in " + path);
            CPDFDocument newDocument = CPDFDocument.InitWithFilePath(path);

            //Validation: try to get the text of the covered area
            string str = newDocument.PageAtIndex(0).GetTextPage().GetSelectText(new CPoint(60, 200), new CPoint(560, 250), new CPoint(0, 0));
            Console.WriteLine("Text in the redacted area is: {0}", str); 
            return true;
        }
    }
}
